hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -1429,6 +1429,8 @@ case class TableMapRows(child: TableIR, newRow: IR) extends TableIR {
   override def partitionCounts: Option[IndexedSeq[Long]] = child.partitionCounts
 
   protected[ir] override def execute(ctx: ExecuteContext): TableValue = {
+    val e = Env[Type]("row" -> child.typ.rowType, "global" -> child.typ.globalType)
```

Comment:
I think we want to back this out in favor of the typechecks you've already added in lowering passes.